### DISCUSSION TITLE
Integrate with R2R for RAG backend

### DISCUSSION THREAD

**Discussion by emrgnt-cmplxty** [jonfairbanks/local-rag#39] [Ideas]
Hi, this is Owen from SciPhi.

I corresponded briefly with @jonfairbanks on r/LocalLlama about integration R2R as a backend provider.

I think this might give local-rag a  more flexible / powerful backend solution and allow the repository to focus more on the application logic. If there is interest I would be happy to help with the integration. I've uploaded our latest demo to make it clearer what R2R does.

https://github.com/jonfairbanks/local-rag/assets/68796651/12846bc2-847d-4ecb-bbbb-3752612e12c0

**Comment by jonfairbanks** [comment]
Hey Owen! Thanks for opening this up. Based on that video, R2R definitely looks powerful. 

I just wrapped a few key items I wanted to get across the line but will definitely spend some time ramping up on this next week.
